packages/mongo/observe_handle.ts

```diff
@@ -4,24 +4,30 @@ let nextObserveHandleId = 1;
 
 export type ObserveHandleCallbackInternal = '_added' | '_addedBefore' | '_changed' | '_movedBefore' | '_removed';
 
+
+export type Callback<T = any> = (...args: T[]) => Promise<void> | void;
+
 /**
  * The "observe handle" returned from observeChanges.
  * Contains a reference to an ObserveMultiplexer.
  * Used to stop observation and clean up resources.
  */
-export class ObserveHandle {
+export class ObserveHandle<T = any> {
   _id: number;
   _multiplexer: ObserveMultiplexer;
   nonMutatingCallbacks: boolean;
   _stopped: boolean;
 
-  _added?: (...args: any[]) => void;
-  _addedBefore?: (...args: any[]) => void;
-  _changed?: (...args: any[]) => void;
-  _movedBefore?: (...args: any[]) => void;
-  _removed?: (...args: any[]) => void;
+  public initialAddsSentResolver: (value: void) => void = () => {};
+  public initialAddsSent: Promise<void>
 
-  constructor(multiplexer: any, callbacks: Record<ObserveHandleCallback, any>, nonMutatingCallbacks: boolean) {
+  _added?: Callback<T>;
+  _addedBefore?: Callback<T>;
+  _changed?: Callback<T>;
+  _movedBefore?: Callback<T>;
+  _removed?: Callback<T>;
+
```

Comment:
Should not be `T[]`? These methods can send a list of args to the callback https://docs.meteor.com/api/collections#callbacks-1